cpp/src/arrow/csv/parser.cc

```diff
@@ -296,6 +297,9 @@ class BlockParserImpl {
     if (UseBulkFilter) {
       const char* bulk_end = RunBulkFilter(parsed_writer, data, data_end, bulk_filter);
       if (ARROW_PREDICT_FALSE(bulk_end == nullptr)) {
+        if (is_final) {
+          final_offset = static_cast<int32_t>(data_end - data);
```

Comment:
Can this simply be
```suggestion
          data = data_end;
```

Reply:
`data` is used in line https://github.com/apache/arrow/blob/master/cpp/src/arrow/csv/parser.cc#L387 if there are fewer rows than expected. Didn't read deep`HandleInvalidRow` code, will updating `data` cause problem?

Reply:
We could perhaps add a test for this fringe situation...

Reply:
Done